File: 4135-concatenate-non-zero-digits-and-multiply-by-sum-i/concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        s = str(n)

        newNumber = 0
        digitSum = 0

        for ch in s:

            if ch != '0':

                digit = int(ch)

                newNumber = newNumber * 10 + digit
                digitSum += digit

        return newNumber * digitSum

File: 4135-concatenate-non-zero-digits-and-multiply-by-sum-i/concatenate-non-zero-digits-and-multiply-by-sum-i.cpp
class Solution {
public:
    long long sumAndMultiply(int n) {

        string s = to_string(n);

        long long newNumber = 0;
        int sum = 0;

        for(char ch : s){

            if(ch != '0'){

                int digit = ch - '0';

                newNumber = newNumber * 10 + digit;

                sum += digit;
            }
        }

        return newNumber * sum;
    }
};